addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -261,13 +365,147 @@ odoo.define('website_slides.quiz', function (require) {
+            } else {
+                $elem.replaceWith(QWeb.render('slide.quiz.new.question.button'));
+            }
+        },
+
+        _displayUpdatedQuestion: function (response) {
+            response.data.element.replaceWith(QWeb.render('slide.quiz.question.view', { question: response.data.newQuestionData }));
+        },
+
+        /**
+         * If the user cancels the creation or update of a Question it resets the display
+         * of the updated Question or it displays back the buttons.
+         *
+         * @param reset
+         * @private
+         */
+        _resetDisplay: function (reset) {
+            if (reset.data.update) {
+                reset.data.$elem.html(reset.data.$oldElem);
+            } else {
+                var $elem = this.$('.o_wslides_js_lesson_quiz_new_question');
+                if (this.quiz.questionsCount > 0) {
+                    $elem.html(QWeb.render('slide.quiz.new.question.button'));
+                } else {
+                    $elem.html(QWeb.render('slide.quiz.new.quiz.button'));
+                }
+            }
+        },
+
+        /**
+         * After deletion of a Question the display is refreshed with the removal of the Question
+         * the reordering of all the remaining Questions and the change of the new Question sequence
+         * if the QuestionFormWidget is initialized.
+         *
+         * @param event
+         * @private
+         */
+        _deleteQuestion: function(event) {
+            var question = event.data;
+            question.remove();
+            this.quiz.questionsCount--;
+            this._reorderQuestions();
+            var $newQuestion = this.$('.o_wslides_js_lesson_quiz_new_question .o_wslides_quiz_question_sequence');
+            $newQuestion.text(parseInt($newQuestion.text()) - 1);
+        },
+    });
+
+    /**
+     * Dialog box shown when clicking the deletion button on a Question.
+     * When confirming it sends a RPC request to delete the Question.
+     */
+    var ConfirmationDialog = Dialog.extend({
+        template: 'slide.quiz.confirm.deletion',
+        xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+            ['/website_slides/static/src/xml/slide_quiz_create.xml']
+        ),
+
+        init: function (parent, options, question) {
+            options = _.defaults(options || {}, {
+                title: _t('Delete Question'),
+                buttons: [
+                    { text: _t('Yes'), classes: 'btn-primary', click: this._confirm },
+                    { text: _t('No'), close: true}
+                ],
+                size: 'medium'
+            });
+            this.question = question;
+            this._super(parent, options);
+        },
+
+        _confirm: function () {
+            var self = this;
+            this._rpc({
+                model: 'slide.question',
+                method: 'unlink',
+                args: [this.question.data('questionId')],
```

Comment:
Passing a whole html element seems overkill.
Maybe just pass the questionId and `self.trigger_up('remove_question', self.questionId); `

The handler of the event will find the appropriate block by its question id and remove it.